### PR TITLE
Add null provider for the privacy API

### PR DESCRIPTION
**Pull request by SammyIsConfused**
Moodle now adheres to the GDPR which means we need to implement a privacy provider. 
See https://docs.moodle.org/dev/Privacy_API#Plugins_which_do_not_store_personal_data for details.  

image

### CODE REVIEW

**Review by ChristianMurphy — APPROVED** [review]

File: lang/en/block_equella_tasks.php [hunk 1]
```diff
@@ -19,5 +19,4 @@
 $string['incompatible'] = 'This version of openEQUELLA is incompatible with this block';
 $string['error'] = 'There has been an error connecting to openEQUELLA.  Will try again in 5 minutes.';
 $string['notasks'] = 'You currently have no tasks';
-
-?>
```
Comment (edalex-ian) [inline lang/en/block_equella_tasks.php:23]: I'm surprised this was happy. You still have the opening `<?php` but then you've chopped this one off. Could be good to return it. :thinking:

File: classes/privacy/provider.php [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+class provider implements \core_privacy\local\metadata\null_provider {
+    // this plugin does not store any personal user data.
+    /**
+     * Get the language string identifier with the component's language
+     * file to explain why this plugin stores no data.
+     */
+    public static function get_reason(): string {
+        return 'privacy:metadata';
+    }
+}
```
Comment (edalex-ian) [inline classes/privacy/provider.php:13]: I'm going to say, if you're opening your file with the php tag (`<?php`) then you should probably end the file closing it (`?>`) to be sure.

**Review by edalex-ian — CHANGES_REQUESTED** [review]
Seems fine, just need to make sure we close those php tags. (You may even want to consider do the same for the other files in this repo.)

Also, same query about the text: https://github.com/openequella/moodle-block_openEQUELLA_links/pull/7#discussion_r477014178

File: classes/privacy/provider.php (hunk 2 above)
Comment (ChristianMurphy) [reply classes/privacy/provider.php:13]: PSR-2, the PHP code style recommends never adding `?>` to pure PHP files.
https://www.php-fig.org/psr/psr-2/#22-files

Comment (edalex-ian) [reply classes/privacy/provider.php:13]: Ah! Okay, cool. If that's what the PHP world likes, I'm happy to go with that. :grin:

**Review by edalex-ian — APPROVED** [review]

**Pull request merged**